### PR TITLE
Compute work order request and response hash as per Trusted compute s…

### PR DESCRIPTION
**Pull request by manojgop**
…pec.

Current implementation of Request and response hash computation has
base64 conversions which is not as per TC spec. This commit fixes this issue.

Signed-off-by: Manoj Gopalakrishnan <manoj.gopalakrishnan@intel.com>

### CODE REVIEW

File: common/crypto_utils/avalon_crypto_utils/crypto_utility.py [hunk 1]
```diff
@@ -252,10 +252,10 @@ def byte_array_to_hex(byte_array):
 
 
 # -----------------------------------------------------------------------------
-def compute_message_hash(message):
+def compute_message_hash(message_bytes):
```
Comment (danintel) [inline common/crypto_utils/avalon_crypto_utils/crypto_utility.py:255]: As long as you are changing this, add a docstring. Something like this (modified from crypto_utils.cpp):

```
"""
Compute SHA256 hash of message_bytes.

Parameters:
    message_bytes Data to hash
Returns:
    binary hash of data
"""
```

File: tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp [hunk 2]
```diff
@@ -308,37 +308,56 @@ namespace tcf {
...
-        ByteArray hash_data;
-        std::string hash_2;
+        ByteArray hash_1 =  tcf::crypto::ComputeMessageHash(
+                                    StrToByteArray(concat_string));
+        // Calculate inData hash
+        ByteArray hash_in_data;
+        std::string hash_in_data_str;
         size_t i;
+        // First sort the inData elements based on index
+        // Sorting is requried to calculate hash deterministically
```
Comment (danintel) [inline tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp:318]: s/requried/required/

File: common/crypto_utils/avalon_crypto_utils/crypto_utility.py (hunk 1 above)
Comment (manojgop) [reply common/crypto_utils/avalon_crypto_utils/crypto_utility.py:255]: Done

File: tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp (hunk 2 above)
Comment (manojgop) [reply tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp]: Done

**Pull request merged**